irohad/main/irohad.cpp

```diff
@@ -85,6 +85,19 @@ DEFINE_bool(wait_for_new_blocks,
             "Startup synchronization policy - waits for new blocks in "
             "blockstore, does not run network");
 
+/**
+ * Legacy configuration precedence: FILE -> ENV
+ * 
+ * Non-legacy configuration precedence: ENV -> FILE
+ * 
+ * Flag is needed so that configuration precedence can be refactored
+ * without it being a breaking change to existing deployments of 1.x
+ */
+DEFINE_bool(legacy_config_precedence,
+            true,
+            "Run with legacy configuration precedence of FILE -> ENV");
```

Comment:
It is great that You described in comments with details, I'd suggest to add some more details to what would be displayed when somebody runs:
```
irohad --help
```
So I suggest to extend:
```
"Run with legacy configuration precedence of FILE -> ENV");
```
into something like that:
```
"Run with legacy configuration precedence of FILE -> ENV. Non-legacy configuration precedence: ENV -> FILE");
```

Reply:
And let's say CONFIG_FILE instead of FILE to be exact

Reply:
@baziorek @kuvaldini Done: bb562c9948902621d93cf3695e3d51865f456238